src/uucore/src/lib/features/checksum.rs

```diff
@@ -75,6 +75,65 @@ struct ChecksumResult {
     pub failed_open_file: i32,
 }
 
+/// Represents a reason for which the processing of a checksum line
+/// could not proceed to digest comparison.
+enum LineCheckError {
+    /// a generic UError was encountered in sub-functions
+    UError(Box<dyn UError>),
+    /// the line is empty or is a comment
+    Skipped,
+    /// the line has a formatting error
+    ImproperlyFormatted,
+    /// the given file is not found, may be a directory
+    CantOpenFile(bool),
+}
+
+impl From<Box<dyn UError>> for LineCheckError {
+    fn from(value: Box<dyn UError>) -> Self {
+        Self::UError(value)
+    }
+}
+
+impl From<ChecksumError> for LineCheckError {
+    fn from(value: ChecksumError) -> Self {
+        Self::UError(Box::new(value))
+    }
+}
+
+/// Represents an error that was encountered when processing a checksum file.
+#[allow(clippy::enum_variant_names)]
+enum FileCheckError {
+    /// a generic UError was encountered in sub-functions
+    UError(Box<dyn UError>),
+    /// the error does not stop the processing of next files
+    NonCriticalError,
+    /// the error must stop the run of the program
+    CriticalError,
```

Comment:
These two variants are quite generic and don't provide much information about what went wrong. When I read the error handler on lines 826 - 830, I have to lookup what is considered a critical error and what not. However, I think it is the job of the error handler to do such a categorization.

Reply:
You're right, I will add more error variants to make it more understandable.